generator/lookml.py

```diff
@@ -0,0 +1,148 @@
+"""Generate lookml from namespaces."""
+import re
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import click
+import yaml
+from google.cloud import bigquery
+from jinja2 import Environment, FileSystemLoader
+
+_env = Environment(
+    loader=FileSystemLoader(
+        (Path(__file__).parent / "templates").absolute().as_posix()
+    ),
+    trim_blocks=True,
+    lstrip_blocks=True,
+)
+
+VIEW_TEMPLATE = _env.get_template("view.lkml")
+
+
+@dataclass
+class Dimension:
+    """LookML View Dimension."""
+
+    path: Tuple[str, ...]
+    field_type: str
+    mode: str
+
+    @property
+    def is_group(self) -> bool:
+        """Determine if this is a dimension_group or dimension."""
+        return not self.hidden and self.type == "time"
+
+    @property
+    def name(self) -> str:
+        """Translate path to a dimension name."""
+        path = self.path
+        # remove suffix for dimension_group with timeframes
+        if self.field_type == "TIMESTAMP":
+            path = *path[:-1], re.sub("_(time|timestamp)$", "", self.path[-1])
+        elif self.field_type == "DATE":
+            path = *path[:-1], re.sub("_date$", "", self.path[-1])
+        # restore suffix if it was the entire pre-parsed field name
+        if path[-1] == "parsed":
+            # undo the above
+            path = self.path
+        return "__".join(path)
+
+    @property
+    def sql(self) -> str:
+        """Translate path to sql."""
+        return "${TABLE}." + ".".join(self.path)
+
+    @property
+    def hidden(self) -> bool:
+        """Determine if this is hidden."""
+        return self.mode == "REPEATED" or self.path == ("document_id",)
```

Comment:
We should also hide `client_ids`. Would be good to move the default list of hidden fields to a constant.